src/main/kotlin/app/revanced/patches/remini/misc/premium/fingerprints/PremiumUnlockFingerprint.kt

```diff
@@ -0,0 +1,18 @@
+package app.revanced.patches.remini.misc.premium.fingerprints
+
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import org.jf.dexlib2.Opcode
+
+object PremiumUnlockFingerprint : MethodFingerprint(
+    opcodes = listOf(
+        Opcode.IGET_BOOLEAN,
+        Opcode.IF_NEZ,
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.MOVE_RESULT_OBJECT,
+        Opcode.IGET_BOOLEAN,
+        Opcode.IF_EQZ,
+        Opcode.GOTO,
+        Opcode.CONST_4
+    ),
+    customFingerprint = { it.parameterTypes == listOf("Ljava/lang/Object;") && it.returnType == "Ljava/lang/Object;" }
```

Comment:
Instead of creating a list every time the custom fingerprint is invoked to check the parameter types, compare the first parameters type. Also there is a dedicated property for return type in `MethodFingerprint`.